lib/validate/clusterconfig.go

```diff
@@ -81,18 +80,21 @@ func ClusterConfiguration(existing, update clusterconfig.Interface) error {
 			return trace.BadParameter("specified service subnet (%v) is the same as existing service subnet",
 				newGlobalConfig.ServiceCIDR)
 		}
-		podSubnet := newGlobalConfig.PodCIDR
-		if podSubnet == "" {
-			podSubnet = globalConfig.PodCIDR
-		}
-		_, podCIDR, err := net.ParseCIDR(podSubnet)
-		if err != nil {
-			return trace.Wrap(err, "invalid pod subnet: %v", podSubnet)
-		}
-		if err := KubernetesSubnets(podCIDR, serviceCIDR); err != nil {
+		serviceCIDRString = newGlobalConfig.ServiceCIDR
+		subnetModified = true
+	}
+
+	if newGlobalConfig.PodSubnetSize != "" {
+		podSubnetSizeString = newGlobalConfig.PodSubnetSize
+		subnetModified = true
+	}
+
+	if subnetModified {
```

Comment:
nit: does it matter if you have modified settings? Even if unmodified, the test should validate fine so there is no danger of running the invalid test. This can simplify a bit.